crates/re_renderer/src/renderer/depth_cloud.rs

```diff
@@ -253,52 +253,53 @@ fn create_and_upload_texture<T: bytemuck::Pod>(
         .textures
         .alloc(&ctx.device, &depth_texture_desc);
 
-    let format_info = depth_texture_desc.format.describe();
-    let width_blocks = depth_cloud.depth_dimensions.x / format_info.block_dimensions.0 as u32;
-    let height_blocks = depth_cloud.depth_dimensions.y / format_info.block_dimensions.1 as u32;
-    let bytes_per_row_unaligned = width_blocks * format_info.block_size as u32;
+    let TextureRowDataInfo {
+        bytes_per_row_unpadded: bytes_per_row_unaligned,
+        bytes_per_row_padded,
+    } = texture_row_data_info(depth_texture_desc.format, depth_texture_desc.size.width);
+
+    // Not supporting compressed formats here.
+    debug_assert!(depth_texture_desc.format.describe().block_dimensions == (1, 1));
 
     // TODO(andreas): CpuGpuWriteBelt should make it easier to do this.
-    let bytes_per_row_aligned =
-        wgpu::util::align_to(bytes_per_row_unaligned, wgpu::COPY_BYTES_PER_ROW_ALIGNMENT);
-    let bytes_padding_per_row = (bytes_per_row_aligned - bytes_per_row_unaligned) as usize;
+    let bytes_padding_per_row = (bytes_per_row_padded - bytes_per_row_unaligned) as usize;
     // Sanity check the padding size. If this happens something is seriously wrong, as it would imply
     // that we can't express the required alignment with the block size.
     debug_assert!(
         bytes_padding_per_row % std::mem::size_of::<T>() == 0,
         "Padding is not a multiple of pixel size. Can't correctly pad the texture data"
     );
-    let blocks_padding_per_row = bytes_padding_per_row / std::mem::size_of::<T>();
+    let num_pixel_padding_per_row = bytes_padding_per_row / std::mem::size_of::<T>();
 
     let mut depth_texture_staging = ctx.cpu_write_gpu_read_belt.lock().allocate::<T>(
         &ctx.device,
         &ctx.gpu_resources.buffers,
-        data.len() + blocks_padding_per_row * height_blocks as usize,
+        data.len() + num_pixel_padding_per_row * depth_texture_desc.size.height as usize,
     );
 
     // Fill with a single copy if possible, otherwise do multiple, filling in padding.
-    if blocks_padding_per_row == 0 {
+    if num_pixel_padding_per_row == 0 {
         depth_texture_staging.extend_from_slice(data);
     } else {
         let row_padding = std::iter::repeat(T::zeroed())
-            .take(blocks_padding_per_row)
+            .take(num_pixel_padding_per_row)
```

Comment:
But in debug this could be horrible slow then :/
Gonna try release in compiler explorer (with a vec instead of the belt buffer), curious if it really optimizes

Reply:
Uh okay this is really messy with a vec https://rust.godbolt.org/z/TYPP4eYdY because of all the alloc checks and inlined allocing. Comparing the snippets I'm inclined to say you're right; the one you suggest ends up being much shorter, that gotta be worth something 😅  (I can't _really_ tell). Going with your suggestion